source/common/config/rds_json.cc

```diff
@@ -156,14 +156,19 @@ void RdsJson::translateRoute(const Json::Object& json_route, envoy::api::v2::Rou
 
   auto* match = route.mutable_match();
 
+  // This is a trick to do a three-way XOR.
+  if ((json_route.hasObject("prefix") + json_route.hasObject("path") +
+       json_route.hasObject("regex")) != 1) {
+    throw EnvoyException("routes must specify one of prefix/path/regex");
```

Comment:
can you check to make sure this case has test coverage? I think it might not currently.

Reply:
I can look into this